cpp/src/arrow/compute/api_scalar.h

```diff
@@ -49,10 +49,44 @@ class ARROW_EXPORT ElementWiseAggregateOptions : public FunctionOptions {
   explicit ElementWiseAggregateOptions(bool skip_nulls = true);
   constexpr static char const kTypeName[] = "ElementWiseAggregateOptions";
   static ElementWiseAggregateOptions Defaults() { return ElementWiseAggregateOptions{}; }
-
   bool skip_nulls;
 };
 
+enum class RoundMode {
```

Comment:
I think it's fine to keep the docstring, and it might also be nice to document the individual enum members (e.g. note that TOWARDS_NEG_INFINITY is also known as 'floor' or 'downward'). Also the comment about the indices is probably redundant - that's the default behavior for C++ and presumably we aren't adding more modes.

Reply:
I see there's a table in compute.rst so maybe a comment mentioning that would be good - at least I know I only grep header files when I'm looking for something.

Reply:
I added comments to individual enums and reference to `compute.rst`.

Reply:
Awesome, thanks.